src/datastore/postgres-store.ts

```diff
@@ -941,26 +945,40 @@ export class PgDataStore
     // 2. Use `pg-cursor` to async read rows from temp table (order by `id` ASC)
     // 3. Drop temp table
     // 4. Close db connection
-    const pg = await this.connect({ usageName: 'get-raw-events', skipMigrations: true });
+    const pg = await this.connect({
+      usageName: 'get-raw-events',
+      skipMigrations: true,
+      withNotifier: false,
+    });
     try {
       const client = await pg.pool.connect();
       try {
         await client.query('BEGIN');
-        await client.query(
-          `
+        await client.query(`
           CREATE TEMPORARY TABLE temp_event_observer_requests(
             id bigint PRIMARY KEY,
             receive_timestamp timestamptz NOT NULL,
             event_path text NOT NULL,
             payload jsonb NOT NULL
           ) ON COMMIT DROP
-          `
-        );
+        `);
+        // Use a `temp_raw_tsv` table first to store the raw TSV data as it might come with duplicate
+        // rows which would trigger the `PRIMARY KEY` constraint in `temp_event_observer_requests`.
+        // We will "upsert" from the former to the latter before event ingestion.
+        await client.query(`
+          CREATE TEMPORARY TABLE temp_raw_tsv
+          (LIKE temp_event_observer_requests)
+          ON COMMIT DROP
+        `);
         onStatusUpdate?.('Importing raw event requests into temporary table...');
-        const importStream = client.query(
-          pgCopyStreams.from(`COPY temp_event_observer_requests FROM STDIN`)
-        );
+        const importStream = client.query(pgCopyStreams.from(`COPY temp_raw_tsv FROM STDIN`));
         await pipelineAsync(readStream, importStream);
+        await client.query(`
+          INSERT INTO temp_event_observer_requests
+          SELECT *
+          FROM temp_raw_tsv
+          ON CONFLICT DO NOTHING;
```

Comment:
I'm looking at how the tsv file could result in duplicate `id`s, since those should just be a mirror of the unique primary key in the `event_observer_requests` table:
https://github.com/hirosystems/stacks-blockchain-api/blob/f4649fd91ce559a14da849757685a7098fcbbf27/src/datastore/postgres-store.ts#L876-L909

It looks like the tsv file can be appended to with the _expected_ `id` before `INSERT INTO event_observer_requests ...` transaction is fully committed, like if the API or pg shutdown after the file write, but before the sql commit finishes. There's probably room to improve the logic there to reduce or eliminate the chance of that happening. But, with the existing behavior, it looks like the _newer_ tsv entries should actually overwrite the older ones on conflict. 

Thoughts?